src/main/java/org/gingesnap/k8s/KubernetesClientProducer.java

```diff
@@ -0,0 +1,41 @@
+package org.gingesnap.k8s;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+import javax.enterprise.inject.Produces;
+import javax.inject.Singleton;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.fabric8.kubernetes.client.ConfigBuilder;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClientBuilder;
+import io.quarkus.arc.lookup.LookupIfProperty;
+
+@Singleton
+public class KubernetesClientProducer {
+
+   private static final Logger log = LoggerFactory.getLogger(KubernetesClientProducer.class);
+
+   @Produces
+   @LookupIfProperty(name = "gingernsnap.k8s.enabled", stringValue = "true")
+   public KubernetesClient kubernetesClient() {
+      log.info("Creating Kubernetes client");
+
+      try {
+         // TODO hardcoded
+         String namespace = Files.readString(Paths.get("/var/run/secrets/kubernetes.io/serviceaccount/namespace"));
+         ConfigBuilder cb = new ConfigBuilder();
+
+         return new KubernetesClientBuilder()
+               .withConfig(cb.withNamespace(namespace).build())
+               .build();
+      } catch (IOException e) {
+         log.error("Unable to determine local namespace", e);
+         return null;
```

Comment:
I think it would be better to throw a runtime exception here as without the namespace it's not possible for the k8s watch to function correctly. This also means that we don't have to use `Instance` in `CacheRuleWatcher`.